### PR TITLE
Create repository methods for cleaning OCR data and changing envelope status

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-452

### Change description ###

- **Create repository methods for cleaning OCR data and changing envelope status.**
- TODO: Create a service that "completes" envelopes (i.e. changes their status to "COMPLETED" and removes OCR data)
- TODO: Feed the service from queue

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepository.java [hunk 1]
```diff
@@ -84,4 +85,12 @@
         @Param("maxFailureCount") int maxFailureCount,
         Pageable pageable
     );
+
+
+    @Modifying(clearAutomatically = true)
+    @Query("update Envelope set status = :status where id = :id")
+    int setEnvelopeStatus(
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepository.java]: how about `updateEnvelopeStatus` to distunguish it from a setter?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepositoryTest.java [hunk 2]
```diff
@@ -60,6 +67,41 @@ public void findEnvelopesToResend_should_filter_based_on_status_and_jurisdiction
...
+        UUID envelope2Id = repo.save(envelope("BULKSCAN", Status.CREATED)).getId();
+
+        repo.setEnvelopeStatus(envelope1Id, Status.NOTIFICATION_SENT);
+        assertEnvelopeHasStatus(envelope1Id, Status.NOTIFICATION_SENT);
+
+        assertEnvelopeHasStatus(envelope2Id, Status.CREATED);
+    }
+
+    @Test
+    public void setEnvelopeStatus_should_return_result_indicating_whether_update_took_place() {
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepositoryTest.java]: how about making it more clear 'should return number of rows updated'?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    public void clearOcrData_should_return_the_number_of_updated_scannable_items() {
+        // given
+        int numberOfScannableItems = 8;
+
+        List<ScannableItem> scannableItems = createScannableItemsWithOcrData(numberOfScannableItems);
+        Envelope envelope = envelope("BULKSCAN", Status.CREATED, scannableItems);
+
+        UUID envelopeId = envelopeRepository.saveAndFlush(envelope).getId();
+
+        // when
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java]: this is 'then' stage

Comment (lgonczar) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java]: this has been split into when and then

**Pull request merged**